### PR TITLE
refactor: move project loading to Install Integration

### PR DESCRIPTION
**Pull request by dionlow**
### Summary
Part 1 of moving loading away from Ampersand Provider
- moved loading down from AmpersandProvider
- move project and integrationList loading to InstallIntegration
- move project loading to ConnectProvider

### CODE REVIEW

File: src/context/ProjectContextProvider.tsx [hunk 1]
```diff
@@ -66,15 +67,19 @@ export function ProjectProvider(
...
+    isLoading,
+  }), [projectIdOrName, project, isLoading]);
 
   return (
     isError(ErrorBoundary.PROJECT, projectIdOrName)
       ? <ErrorTextBox message={`Error loading project ${projectIdOrName}`} />
       : (
         <ProjectContext.Provider value={contextValue}>
-          {isLoading ? <LoadingCentered /> : children}
+          {children}
```
Comment (dionlow) [inline src/context/ProjectContextProvider.tsx:82]: removes loading from Project Provider

File: src/context/IntegrationListContextProvider.tsx [hunk 2]
```diff
@@ -59,15 +60,15 @@ export function IntegrationListProvider(
...
-  }), [integrations]);
+    integrations, isLoading,
+  }), [integrations, isLoading]);
 
   return (
     isError(ErrorBoundary.INTEGRATION_LIST, projectIdOrName)
       ? <ErrorTextBox message="Error retrieving integrations for the project, double check the API key" />
       : (
         <IntegrationListContext.Provider value={contextValue}>
-          {isLoading ? <LoadingCentered /> : children}
```
Comment (dionlow) [inline src/context/IntegrationListContextProvider.tsx:70]: removes UI Loading component from IntegrationList Provider

File: src/context/ConnectionsContextProvider.tsx [hunk 3]
```diff
@@ -90,12 +90,16 @@ export function ConnectionsProvider({
     setSelectedConnection,
   }), [connections, selectedConnection, setConnections, setSelectedConnection]);
 
+  if (isLoading || isProjectLoading) {
```
Comment (dionlow) [inline src/context/ConnectionsContextProvider.tsx:93]: catch project loading at top level of ConnectProvider

File: src/components/Configure/InstallIntegration.tsx [hunk 4]
```diff
@@ -63,10 +66,15 @@ export function InstallIntegration(
     onUninstallSuccess, fieldMapping,
   }: InstallIntegrationProps,
 ) {
-  const { projectId } = useProject();
+  const { projectId, isLoading: isProjectLoading } = useProject();
+  const { isLoading: isIntegrationListLoading } = useIntegrationList();
   const { errorState } = useErrorState();
   const { seed, reset } = useForceUpdate();
```
Comment (dionlow) [inline src/components/Configure/InstallIntegration.tsx:73]: catch project and integration list loading at top of InstallIntegration

File: src/context/ProjectContextProvider.tsx [hunk 5]
```diff
@@ -17,13 +16,15 @@ interface ProjectContextValue {
...
   projectIdOrName: string;
+  isLoading: boolean;
 }
 
 export const ProjectContext = createContext<ProjectContextValue>({
   project: null,
   appName: '',
   projectId: '',
   projectIdOrName: '',
+  isLoading: false,
```
Comment (laurenzlong) [inline src/context/ProjectContextProvider.tsx:27]: shouldn't the default value for `isLoading` be true?

**Review by laurenzlong — APPROVED** [review]

File: src/context/ProjectContextProvider.tsx (hunk 5 above)
Comment (dionlow) [reply src/context/ProjectContextProvider.tsx]: switched.

**Pull request merged**